resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ContentDescriptionManager.java

```diff
@@ -65,6 +73,10 @@
  * @see IFile#getContentDescription()
  */
 public class ContentDescriptionManager implements IManager, IRegistryChangeListener, IContentTypeManager.IContentTypeChangeListener, ILifecycleListener {
+	private static final String EMPTY = ""; //$NON-NLS-1$
+	private static final String DELIMITER = System.lineSeparator();
+	private static final String SPACE = " "; //$NON-NLS-1$
```

Comment:
One nitpicky design-related comment: these constants are a bit confusing, as they do not belong to the `ContentDescriptionManager` but to some kind of `PlatformStateDescription` that is handled by `getCurrentPlatformState()`, `getCachedPlatformState()` and the property setter in `shutdown()`. In order to separate concerns and clarify where these constants belong to, you could encapsulate this logic in some (inner, static) `PlatformStateDescription` class.

Reply:
Well actually this just to not make the code scattered with all these `//$NON-NLS-1$` stuff... I can't get why one ever would want to "translate" an empty String or a single space but I'm sure there is a reason I just don't understand :-P